src/ts.ts

```diff
@@ -0,0 +1,83 @@
+import fs from "fs";
+import path from "path";
+import importCwd from "import-cwd";
+import type { LanguageService, LanguageServiceHost, ParseConfigFileHost, Program } from "typescript";
+
+function isTypeScriptVersionSupported(major: number, minor: number) {
+  if (major < 3) return false;
+  if (major < 4) return minor >= 5;
+  return true;
+}
+
+type TypeScript = typeof import("typescript");
+
+export function loadTypeScript(): TypeScript {
+  try {
+    const ts = importCwd("typescript") as typeof import("typescript");
+    const [major, minor] = ts.versionMajorMinor.split('.');
+    if (isTypeScriptVersionSupported(parseInt(major, 10), parseInt(minor, 10))) {
+      return ts;
+    }
+  } catch {}
+  return require("typescript");
```

Comment:
Tries to load TypeScript from the current working directory, falls back to the bundled version if the local version isn’t a supported version or if it’s not present.